datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestStickyPartitionAssignment.java

```diff
@@ -58,11 +62,20 @@ public void setup() throws IOException {
     String zkConnectionString = _embeddedZookeeper.getConnection();
     _embeddedZookeeper.startup();
     _zkClient = new ZkClient(zkConnectionString);
+    DynamicMetricsManager.createInstance(new MetricRegistry(), "TestStickyPartitionAssignment");
   }
 
   @AfterMethod
-  public void teardown() {
+  public void teardown() throws Exception {
     _embeddedZookeeper.shutdown();
+    // A hack to force clean up DynamicMetricsManager
```

Comment:
Can you please explain the reason behind the hack to understand the rationale for future use? I'm not able to make sense from this code.

Reply:
I got this from another test, so don't know the rationale, but I do remember there was an issue with some tests getting older metrics, and I think this was added to remove that. I can double check what this is all about if you'd like.